xbmc/video/jobs/VideoLibraryRefreshingJob.cpp

```diff
@@ -94,18 +95,32 @@ bool CVideoLibraryRefreshingJob::Work(CVideoDatabase &db)
   bool failure = false;
   do
   {
+    std::unique_ptr<CVideoInfoTag> pluginTag;
+    std::unique_ptr<CGUIListItem::ArtMap> pluginArt;
+
     if (!ignoreNfo)
     {
       std::unique_ptr<IVideoInfoTagLoader> loader;
       loader.reset(CVideoInfoTagLoaderFactory::CreateLoader(*m_item, scraper,
-                                                            scanSettings.parent_name_root));
+                                                            scanSettings.parent_name_root, m_forceRefresh));
       // check if there's an NFO for the item
       CInfoScanner::INFO_TYPE nfoResult = CInfoScanner::NO_NFO;
       if (loader)
       {
-        CVideoInfoTag dummy;
-        nfoResult = loader->Load(dummy, false);
-        if (nfoResult == CInfoScanner::URL_NFO)
+        std::unique_ptr<CVideoInfoTag> tag(new CVideoInfoTag());
+        nfoResult = loader->Load(*tag, false);
+        if (nfoResult == CInfoScanner::FULL_NFO && m_item->IsPlugin() && scraper->ID() == "metadata.local")
+        {
+          // get video info and art from plugin source with metadata.local scraper
+          if (scraper->Content() == CONTENT_TVSHOWS && !m_item->m_bIsFolder && tag->m_iIdShow < 0)
+            // preserve show_id for episode
+            tag->m_iIdShow = m_item->GetVideoInfoTag()->m_iIdShow;
+          pluginTag = std::move(tag);
+          CVideoTagLoaderPlugin* nfo = dynamic_cast<CVideoTagLoaderPlugin*>(loader.get());
```

Comment:
why the need to cast, can't you just pass the art down through load ?

Reply:
Load(CVideoInfoTag& tag, bool prioritise, std::vector<EmbeddedArt>* art) is IVideoInfoTagLoader interface method which I can not change and "art" param here is not what we need (CGUIListItem::ArtMap).